tests/functional/algos/test_algos.py

```diff
@@ -247,3 +247,64 @@ def test_with_evc(algorithm):
         param = best_trial._params[1]
         assert param.name == "x"
         assert param.type == "real"
+
+
+@pytest.mark.parametrize(
+    "algorithm", algorithm_configs.values(), ids=list(algorithm_configs.keys())
+)
+def test_parallel_workers(algorithm, tmp_path):
+    """Test parallel execution with joblib"""
+    pickle_host = str(tmp_path / "db.pkl")
+
+    name = "{}_exp".format(algorithm)
+
+    storage = {"type": "legacy", "database": {"type": "PickledDB", "host": pickle_host}}
+    base_exp = create_experiment(
+        name=name,
+        space=space_with_fidelity,
+        algorithms=algorithm_configs["random"],
+        storage=storage,
+    )
+    base_exp.workon(rosenbrock, max_trials=10)
+
+    exp = create_experiment(
```

Comment:
We can avoid tests with evc when testing multiple workers. Unless you see a potential issue with EVC that is specific to workon with n_workers > 1?

Reply:
You know what, I actually did not notice the issue of EVC at the beginning, I did this just because I think with or without n_workers >1, the behavior should be the same. But just when I re-ran the case, it failed, the root cause is in last commit, I only restore the storage for the current experiment node after de-serialize the object, but did not restore the storage for parent nodes, this is really weird. Any way, after I fixed the issue, it works.

Reply:
haha I am glad this test caught it! EVC will never stop to surprise us. :sweat_smile: This was probably due to the serialization of the nodes in the EVC tree. Dropping the other nodes would have done the job if I am right. But your solution looks simpler. :)

Reply:
So we should keep the tests with EVC indeed.